hail/src/main/scala/is/hail/expr/types/physical/PBaseStruct.scala

```diff
@@ -174,6 +174,16 @@ abstract class PBaseStruct extends PType {
     region.setBit(offset, missingIdx(fieldIdx))
   }
 
+  def setFieldPresent(region: Region, offset: Long, fieldIdx: Int) {
+    assert(!fieldRequired(fieldIdx))
```

Comment:
Why does the field need to not be required?

Reply:
if the field is required, then I think we don't allocate missingness bits. (I'm not sure)